compiler/hash-exhaustiveness/src/storage.rs

```diff
@@ -1,13 +1,14 @@
 //! Stores [DeconstructedPat]s and [DeconstructedCtor]s.'
 use hash_storage::{new_store_key, store::DefaultStore};
+use hash_tir::node::Node;
 
 use crate::{construct::DeconstructedCtor, deconstruct::DeconstructedPat};
 
 new_store_key!(pub DeconstructedPatId, derives = Debug);
-pub type DeconstructedPatStore = DefaultStore<DeconstructedPatId, DeconstructedPat>;
+pub type DeconstructedPatStore = DefaultStore<DeconstructedPatId, Node<DeconstructedPat>>;
 
 new_store_key!(pub DeconstructedCtorId, derives = Debug);
-pub type DeconstructedCtorStore = DefaultStore<DeconstructedCtorId, DeconstructedCtor>;
+pub type DeconstructedCtorStore = DefaultStore<DeconstructedCtorId, Node<DeconstructedCtor>>;
```

Comment:
Not sure this is right to do, there isn't really s need for them to be in nodes.

Reply:
It is so that the recovered `TIR` nodes can have their original origins

Reply:
But they don’t need an origin, its only ever used in a position so that it can be printed, the locations are irrelevant

Reply:
I guess, though it might be useful for richer errors, for example to justify why there are/aren't more constructors it can point to the constructors definition.. I could also remove it if you think that is better..

Reply:
that is impossible since the ones that will be generated will always have an un-trackable span

Reply:
I think we can remove it for now until there is a more explicit intent on using the information. If it is later deemed that the information is useful, we can re-add it. Perhaps we should save those changes as a patch somewhere?